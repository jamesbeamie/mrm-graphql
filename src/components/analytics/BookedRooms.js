/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';
import TableHead from '../helpers/TableHead';
import '../../assets/styles/bookedroom.scss';
import { pollRed } from '../../utils/images/images';
import Tip from '../commons/Tooltip';
import ErrorIcon from '../commons/ErrorIcon';
import Overlay from '../commons/Overlay';

/**
 *
 * Component for showing booked rooms
 *
 * @param {Object} bookedRoomsObject
 *
 * @returns {JSX}
 */
const BookedRooms = ({
  bookedRoomText, bookedRoomsList, fetching, tip,
}) => (
  <div className="wrap-booked-room overlay-container">
    {fetching && <Overlay />}
    <div className="booked-room-header">
      <img src={pollRed} alt="Pull" />
      <h4>{bookedRoomText}</h4>
      {Tip(tip)}
    </div>
    <div className="booked-room-list">
      <div className="table">
        <TableHead titles={['Room', 'Meetings', '% Share of All Meetings']} />
        {bookedRoomsList.length === 0 && !fetching ? (
          <div className="table__body">
            {<ErrorIcon message={bookedRoomsList.length === 0 && 'No resource found'} />}
          </div>
        ) : (
          <div className="table__body">
            {
              (fetching ? [{
                roomName: 'Room name',
                meetings: 'Meetings',
                percentage: '',
              }] : bookedRoomsList).map((room, index) => (
                <div className="table__row--analytics" key={index}>
                  <span>{room.roomName}</span>
                  <span>{room.numberOfBookings}</span>
                  <span>{Math.round(room.bookingsPercentageShare)}%</span>
                </div>
              ))
            }
          </div>
        )}
      </div>
    </div>
  </div>
);

BookedRooms.propTypes = {
  bookedRoomText: PropTypes.string.isRequired,
  tip: PropTypes.string.isRequired,
  bookedRoomsList: PropTypes.instanceOf(Array),
  fetching: PropTypes.bool.isRequired,
};

BookedRooms.defaultProps = {
  bookedRoomsList: [],
};
export default BookedRooms;
