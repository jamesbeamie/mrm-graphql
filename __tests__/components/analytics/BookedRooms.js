import React from 'react';
import { shallow, mount } from 'enzyme';
import BookedRooms from '../../../src/components/analytics/BookedRooms';
import roomUsage from '../../../__mocks__/rooms/mostUsedRooms';

describe('BookedRooms component', () => {
  const props = {
    bookedRoomsList: roomUsage,
    pollIcon: '',
    moreIcon: '',
    bookedRoomText: '',
    fetching: false,
    tip: '',
  };

  const wrapper = shallow(<BookedRooms {...props} />);
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(' should render rooms data in  Rooms table', () => {
    const tableWrapper = mount(<BookedRooms {...props} />);
    expect(
      tableWrapper
        .find('.table').length,
    ).toBe(1);
    expect(
      tableWrapper
        .find('.table__row--analytics').length,
    ).toBe(2);
  });

  it('should render error div', () => {
    const bookedRoomsComponent = mount(
      <BookedRooms
        {...props}
        bookedRoomsList={[]}
      />,
    );
    expect(bookedRoomsComponent.find('ErrorIcon').exists()).toBeTruthy();
  });

  it('should load ProgressBar div', () => {
    const bookedRoomsComponent = mount(<BookedRooms {...props} bookedRoomsList={[]} fetching />);
    expect(bookedRoomsComponent.find('Overlay')).toHaveLength(1);
  });
});
