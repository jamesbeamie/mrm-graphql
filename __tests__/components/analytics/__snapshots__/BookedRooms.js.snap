// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookedRooms component should render correctly 1`] = `
<div
  className="wrap-booked-room overlay-container"
>
  <div
    className="booked-room-header"
  >
    <img
      alt="Pull"
      src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fpoll_red.svg?alt=media&token=f162d826-43a7-4bcb-aac8-0077ea6a4885"
    />
    <h4 />
    <ThemedTooltippedComponent
      className=""
      composeTheme="deeply"
      mapThemrProps={[Function]}
      theme={null}
      tooltip=""
      tooltipDelay={0}
      tooltipHideOnClick={true}
      tooltipPosition="top"
      tooltipShowOnClick={true}
    >
      <img
        alt="help icon"
        src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fhelp_outline_24px.svg?alt=media&token=839e3137-c29e-463d-9a68-361e28794911"
      />
    </ThemedTooltippedComponent>
  </div>
  <div
    className="booked-room-list"
  >
    <div
      className="table"
    >
      <TableHead
        titles={
          Array [
            "Room",
            "Meetings",
            "% Share of All Meetings",
          ]
        }
      />
      <div
        className="table__body"
      >
        <div
          className="table__row--analytics"
          key="0"
        >
          <span>
            Kampala
          </span>
          <span />
          <span>
            NaN
            %
          </span>
        </div>
        <div
          className="table__row--analytics"
          key="1"
        >
          <span>
            Gulu
          </span>
          <span />
          <span>
            NaN
            %
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
